fcrepo-http-commons/src/main/java/org/fcrepo/http/commons/responses/ViewHelpers.java

```diff
@@ -146,15 +150,30 @@ public String getVersionSubjectUrl(final UriInfo uriInfo, final Node subject) {
         return null;
      }
 
+    /**
+     * Get the date time as the version label.
+     *
+     * @param graph the graph
+     * @param subject the subject
+     * @return the datetime in RFC 1123 format.
+     */
+    public String getVersionLabel(final Graph graph, final Node subject) {
+        final Instant datetime = getVersionDate(graph, subject);
+        return RFC_1123_DATE_TIME.withZone(ZoneId.of("UTC")).format(datetime);
+    }
+
     /**
      * Gets a modification date of a subject from the graph
      *
      * @param graph the graph
      * @param subject the subject
      * @return the modification date if it exists
      */
-    public Optional<String> getVersionDate(final Graph graph, final Node subject) {
-        return getValue(graph, subject, CREATED_DATE.asNode());
+    public Instant getVersionDate(final Graph graph, final Node subject) {
+        final String[] pathParts = subject.getURI().split("/");
+        final Instant datetime = DateTimeFormatter.ofPattern("yyyyMMddHHmmss").withZone(ZoneId.of("UTC"))
```

Comment:
I wonder if we should make this pattern ("yyyyMMddHHmmss") a static variable for reuse, since it represents part of a MEMENTO's id and it's used in a few spots?     (fcrepo-kernel-modeshape -> VersionServiceImpl.java and some tests)

Reply:
Sure I can do that.